src/backend/XmlHttpRequest.ts

```diff
@@ -31,6 +33,10 @@ export interface XmlHttpRequestOptions {
   // Used as the URL prefix for fetched files.
   // Default: Fetch files relative to `url`.
   baseUrl?: string;
+  // Whether to use Fetch or XmlHttpRequest.
+  // Can be 'xhr', 'fetch', or 'auto'.
+  // Default: 'auto' which uses fetch if its available, and XHR as a fallback
```

Comment:
Yeah I hadn't even thought of it, but in Node someone might want to use `http`. I don't see *why* -
 I would just use `isomorphic-fetch` in that scenario - but I'm sure there are cases. Maybe to enable streaming? Or proxying. Or performance reasons or something.

Yes, we have an abstraction conundrum. One solution with two cases, or two solutions? If we separate XmlHttpRequest and FetchFS, then the user will be responsible for handling it, and have to learn and adapt to the oddities like: there's no XmlHttpRequest in ServiceWorker *but* Fetch doesn't have a synchronous option. I think having both together makes it easiest on the user. There are not so many possibilities that we can't provide an optimal experience for every use case. I like the current behavior, which lets you use sync or async file system operations, but uses fetch when it can. So in Service Workers sync operations will not be supported, but that's the only catch.